public/app/features/canvas/runtime/element.tsx

```diff
@@ -445,10 +445,19 @@ export class ElementState implements LayerElement {
   };
 
   render() {
-    const { item } = this;
+    const { item, div } = this;
+    const scene = this.getScene();
+    // TODO: Rethink selected state handling
```

Comment:
😬 this is the piece I am a bit nervous about fixing later 

does the `div` have any attributes that can be inspected? (also not great, but better than the tree walking and scanning with every render)

Reply:
With selecto we could in theory add a class to the selected element(s) divs and remove it on un-selection (see [example here](https://daybrush.com/selecto/release/latest/doc/Selecto.html#.event:select)). Not sure this helps us too much though given that adds an additional layer of complexity vs. just getting current selection state from selecto itself.

> does the `div` have any attributes that can be inspected? (also not great, but better than the tree walking and scanning with every render)

This current implementation doesn't tree walk / scan with every render as it just retrieves a set property on the selecto object (so in theory pretty minimal performance hit) (see implementation details of `getSelectedTargets` [here](https://daybrush.com/selecto/release/latest/doc/packages_selecto_src_SelectoManager.tsx.html#line229))

Is our concern that the element should know / have control over its own selected state vs. just managing it through selecto? I think the current approach is viable, at least for now, though I could be missing something. Open to discussing this more though

Reply:
it is fine for now... but selection handling feels like the most fragile/finicky bits 😬